noir-projects/noir-protocol-circuits/crates/rollup-lib/src/block_root/block_root_rollup_inputs.nr

```diff
@@ -0,0 +1,146 @@
+        components::assert_prev_rollups_follow_on_from_each_other(left, right);
+
+        // Insert subtree into the l1 to l2 data tree
+        let empty_l1_to_l2_subtree_root = calculate_empty_tree_root(L1_TO_L2_MSG_SUBTREE_HEIGHT);
+        let new_l1_to_l2_message_tree_snapshot = append_only_tree::insert_subtree_to_snapshot_tree(
+            self.start_l1_to_l2_message_tree_snapshot,
+            self.l1_to_l2_message_subtree_sibling_path,
+            empty_l1_to_l2_subtree_root,
+            self.l1_to_l2_roots.public_inputs.converted_root,
+            // TODO(Kev): For now we can add a test that this fits inside of
+            // a u8.
+            L1_TO_L2_MSG_SUBTREE_HEIGHT as u8
+        );
+
+        let state = StateReference { l1_to_l2_message_tree: new_l1_to_l2_message_tree_snapshot, partial: right.end };
+
+        let content_commitment = ContentCommitment {
+            num_txs: (left.num_txs + right.num_txs) as Field,
+            txs_effects_hash: components::compute_txs_effects_hash(self.previous_rollup_data),
+            in_hash: self.l1_to_l2_roots.public_inputs.sha_root,
+            out_hash: components::compute_out_hash(self.previous_rollup_data)
+        };
+
+        let total_fees = components::accumulate_fees(left, right);
+
+        let header = Header {
+            last_archive: left.constants.last_archive,
+            content_commitment,
+            state,
+            global_variables: left.constants.global_variables,
+            total_fees
+        };
+
+        // Build the block hash for this by hashing the header and then insert the new leaf to archive tree.
+        let block_hash = header.hash();
+
+        // Update the archive
+        let archive = append_only_tree::insert_subtree_to_snapshot_tree(
+            self.start_archive_snapshot,
+            self.new_archive_sibling_path,
+            0,
+            block_hash,
+            0
+        );
+
+        let mut fee_arr = [FeeRecipient::empty(); 32];
+        fee_arr[0] = FeeRecipient { recipient: left.constants.global_variables.coinbase, value: total_fees };
+
+        // RootRollupPublicInputs { archive, header, vk_tree_root }
+        BlockRootOrBlockMergePublicInputs {
+            previous_archive: left.constants.last_archive, // archive before this block was added
+            new_archive: archive, // archive once this block was added
+            previous_block_hash: self.previous_block_hash,
+            end_block_hash: block_hash, // current newest block hash = this block hash
+            start_global_variables: left.constants.global_variables, // we have asserted that left.constants == right.constants => ...
+            end_global_variables: left.constants.global_variables, // with a current block range of 1, we only have 1 set of constants
+            out_hash: content_commitment.out_hash,
+            fees: fee_arr,
+            vk_tree_root: left.constants.vk_tree_root
+        }
```

Comment:
This file is mostly just a renamed version of current `root_rollup_inputs`, with these outputs as the main change.